cpp/src/arrow/dataset/file_ipc.cc

```diff
@@ -197,8 +197,8 @@ std::shared_ptr<FileWriteOptions> IpcFileFormat::DefaultWriteOptions() {
   std::shared_ptr<IpcFileWriteOptions> ipc_options(
       new IpcFileWriteOptions(shared_from_this()));
 
-  ipc_options->options =
-      std::make_shared<ipc::IpcWriteOptions>(ipc::IpcWriteOptions::Defaults());
+  ipc_options->properties =
+      std::make_shared<ipc::feather::WriteProperties>(ipc::feather::WriteProperties::Defaults());
```

Comment:
The R failures come (at least partially) from this rename, so we'ld need to update the name to properties there if we're going to do it as well.